### ISSUE TITLE
When there are two windows open in one workspace, pressing CTRL+Q to kill one will crash SwayFX

### ISSUE THREAD

**Issue by AjiBuster499** [WillPower3309/swayfx#72]
### Please read the following before submitting:
- Proprietary graphics drivers, including nvidia, are not supported. Please use the open source equivalents, such as nouveau, if you would like to use Sway.
- Please do NOT submit issues for information from the github wiki. The github wiki is community maintained and therefore may contain outdated information, scripts that don't work or obsolete workarounds.
  If you fix a script or find outdated information, don't hesitate to adjust the wiki page.

Since I had to crash SwayFX several times for this report, I'm writing the actual report from a different computer. Some information that's not in a gist may have typos, as I can't just copy-paste between two computers 😅 . Apologies in advance if such typos occur.

### Please fill out the following:
- **Swayfx Version:** 0.1.1
  - `sway -v` sway version 0.1-e8fa6cd (Dec 14 2022, branch 'master')

- **Debug Log:**
  - Run `sway -d 2> ~/sway.log` from a TTY and upload it to a pastebin, such as gist.github.com.
  - This will record information about sway's activity. Please try to keep the reproduction as brief as possible and exit sway.
  - Attach the **full** file, do not truncate it.
Since I could not reproduce with a minimal config, I also added in a quick test using my regular config. [Here](https://gist.github.com/AjiBuster499/7d3b128cb8e943c1ef17d85aa00130b5) is the gist containing the logs of both regular and minimal.

- **Configuration File:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.
For this testing, I used the default configuration with two changes: I changed `$term` to kitty, and `$menu` to wofi, as I use these over the defaults, and needed a way to open application windows. [Here](https://gist.github.com/AjiBuster499/2acf026750f28b1bb8283d5232aff560) are the configs used, both regular and minimal.

- **Stack Trace:**
  - This is only needed if sway crashes.
  - If you use systemd, you should be able to open the coredump of the most recent crash with gdb with
    `coredumpctl gdb sway` and then `bt full` to obtain the stack trace.
  - If the lines mentioning sway or wlroots have `??` for the location, your binaries were built without debug symbols. Please compile both sway and wlroots from source and try to reproduce.
 [Here](https://gist.github.com/AjiBuster499/073c8263db051804318fe10ad054713fl) is the gist of the last core dump.

- **Description:**
  - The steps you took in plain English to reproduce the problem.
Step 1: Open SwayFX
Step 2: Open two applications that can be closed with CTRL+Q (did not test on more than two)
Step 3: Press CTRL+Q to kill one
Step 4: Witness crash.

**Comment by ErikReider** [comment]
The gist is not public or the URL is broken?

What applications did you test with? I tested with Firefox and it worked as expected.

Lastly, is it reproducible with sway 1.7?

**Comment by RochaLAJ** [comment]
Same is happening with me!
I tested with Kitty. Two terminals side by side produce the same crash once i type "exit" or anything. I removed the bind to exit from the config file because i thought that my keyboard was hanging some key but the problem still persist.

**Comment by WillPower3309** [comment]
Thanks for the report. I'll get this fixed ASAP

**Comment by WillPower3309** [comment]
Reverted most recent version to 0.1 until this is resolved

**Comment by AjiBuster499** [comment]
> The gist is not public or the URL is broken?
Sorry, first time using gists, made them public.
> What applications did you test with? I tested with Firefox and it worked as expected.
I noticed it initially with Thunderbird and Discord, additionally, I also had it crash when multiple instances of Firefox were open. For the tests I believe I used NetworkManager and PCManFM.
> Lastly, is it reproducible with sway 1.7?
No, it is not.

**Comment by WillPower3309** [comment]
Should be fixed now, needed to remove SMART_GAPS support as it was causing the issue. Will investigate how to bring smart gaps back without any issues like this

**Comment by WillPower3309** [comment]
Please let me know if it persists

**Comment by AjiBuster499** [comment]
Hi, using `sway-git` on the AUR, I can't reproduce the issue currently. Tested with my regular config, with both Discord/Thunderbird, and 3 instances of Firefox, and no crashes. Thanks for the fix and hope the smart_gaps rounded corners can come back soon, as I think it's a nice feature to have too!
